logsrvd/sendlog.c

```diff
@@ -85,10 +100,78 @@ help(void)
 	"  -i, --iolog_id           remote ID of I/O log to be resumed\n"
 	"  -p, --port               port to use when connecting to host\n"
 	"  -r, --restart            restart previous I/O log transfer\n"
+#if defined(HAVE_OPENSSL)
+	"  -t, --tls                set the communication over TLS with the audit server\n"
+	"  -b, --ca-bundle          certificate bundle file to verify server's cert against\n"
+	"  -c, --cert               certificate file for TLS handshake\n"
+	"  -k, --key                private key file\n"
+#endif
 	"  -V, --version            display version information and exit\n"));
     exit(0);
 }
 
+#if defined(HAVE_OPENSSL)
+static SSL_CTX *
+init_tls_client_context(const char *ca_bundle_file, const char *cert_file, const char *key_file)
+{
+    const SSL_METHOD *method;
+    SSL_CTX *ctx;
+
+    debug_decl(init_tls_client_context, SUDO_DEBUG_UTIL)
+
+    SSL_library_init();
+    OpenSSL_add_all_algorithms();
+    SSL_load_error_strings();
+
+    if ((method = TLS_client_method()) == NULL) {
+        sudo_debug_printf(SUDO_DEBUG_ERROR|SUDO_DEBUG_LINENO,
+            "creation of SSL_METHOD failed: %s",
+            ERR_error_string(ERR_get_error(), NULL));
+        goto bad;
+    }
+    if ((ctx = SSL_CTX_new(method)) == NULL) {
+        sudo_debug_printf(SUDO_DEBUG_ERROR|SUDO_DEBUG_LINENO,
+            "creation of new SSL_CTX object failed: %s",
+            ERR_error_string(ERR_get_error(), NULL));
+        goto bad;
+    }
+
+    if (cert_file) {
+        if (!SSL_CTX_use_certificate_file(ctx, cert_file, X509_FILETYPE_PEM)) {
```

Comment:
Should this be SSL_CTX_use_certificate_chain_file() instead?

Reply:
fixed